src/applications/vaos/actions/expressCare.js

```diff
@@ -115,48 +171,70 @@ export function fetchExpressCareWindows() {
   };
 }
 
-export function routeToPageInFlow(flow, router, current, action) {
+/**
+ * Fetches request limits for all active windows and selects the first one
+ * where the user has not reached their request limit
+ */
+export function fetchRequestLimits() {
   return async (dispatch, getState) => {
     dispatch({
-      type: FORM_PAGE_CHANGE_STARTED,
+      type: FORM_FETCH_REQUEST_LIMITS,
     });
 
-    const nextAction = flow[current][action];
-    let nextPage;
+    try {
+      const activeFacilityIds = selectActiveExpressCareWindows(
+        getState(),
+        moment(),
+      ).map(window => window.facilityId);
+
+      const promiseIds = [...activeFacilityIds];
+      const requestLimits = [];
+
+      // Temporarily limit concurrent calls to 5 at at time while we
+      // wait for a new endpoint that will accept multiple facilityIds
+      while (promiseIds.length) {
+        requestLimits.push(
+          // eslint-disable-next-line no-await-in-loop
+          await Promise.all(
+            promiseIds
+              .splice(0, 5)
+              .map(facilityId => getRequestLimits(facilityId, EXPRESS_CARE)),
+          ),
+        );
+      }
 
-    if (typeof nextAction === 'string') {
-      nextPage = flow[nextAction];
-    } else {
-      const nextStateKey = await nextAction(getState(), dispatch);
-      nextPage = flow[nextStateKey];
-    }
+      const eligibleFacility = []
+        .concat(...requestLimits)
+        .map((limit, index) => {
+          const facilityId = activeFacilityIds[index];
+          return {
+            facilityId,
+            siteId: facilityId.substring(0, 3),
+            ...limit,
+          };
+        })
+        .find(limit => limit.numberOfRequests < limit.requestLimit);
+
+      const isUnderRequestLimit = !!eligibleFacility;
 
-    if (nextPage?.url) {
       dispatch({
-        type: FORM_PAGE_CHANGE_COMPLETED,
+        type: FORM_FETCH_REQUEST_LIMITS_SUCCEEDED,
+        facilityId: eligibleFacility?.facilityId || null,
+        siteId: eligibleFacility?.siteId || null,
+        isUnderRequestLimit,
       });
-      router.push(nextPage.url);
-    } else if (nextPage) {
-      throw new Error(`Tried to route to a page without a url: ${nextPage}`);
-    } else {
-      throw new Error('Tried to route to page that does not exist');
+
+      return isUnderRequestLimit;
```

Comment:
Return this value to `expressCareFlow.js` so it knows whether or not to go to the next page.